akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1197,7 +1197,10 @@ private[stream] object Collect {
             case Some(v) ⇒
               // #20217 the future is already here, avoid scheduling it on the dispatcher
               holder.setElem(v)
-              pushNextIfPossible()
+              v match {
+                case Failure(ex) if holder.supervisionDirectiveFor(decider, ex) == Supervision.Stop ⇒ failStage(ex)
```

Comment:
Through which path would have the decider been called twice?

Reply:
Uff. Memory is foggy but I think it was called once to fail fast if the decider decides stop in the `futureCB`, and then again in `pushNextIfPossible`, see repeating tests added in the original PR